src/ok-computer.ts

```diff
@@ -93,15 +93,29 @@ export const isError = <Err>(err: Err): err is Exclude<Err, undefined> =>
   !!listErrors(err).length;
 export const hasError = isError;
 
+export interface AssertErrParams<Err> {
+  readonly error: Err;
+  readonly errorList: ErrItem<unknown>[];
+}
+
 export function assert<V extends Validator>(
-  validator: V,
   value: unknown,
-  logValue: boolean = false
+  validator: V,
+  err:
+    | Error
+    | string
+    | ((params: AssertErrParams<ExtractErr<V>>) => Error | string) = ({
+    errorList
+  }) => new AssertError(errorList)
 ): asserts value is Infer<V> {
-  const errors = listErrors(validator(value));
-  if (errors.length) {
-    throw new AssertError(errors, logValue ? value : undefined);
+  const error = validator(value);
+  const errorList = listErrors(error);
+  if (!errorList.length) {
+    return;
   }
+  const result =
+    typeof err === 'function' ? err({ error: error as any, errorList }) : err;
+  throw typeof result === 'string' ? new Error(result) : result;
```

Comment:
If you want to log values, you can do that with `assert(value, string, ({ errorList }) => new AssertError(errorList, value))`